LoopKitUI/Extensions/Environment+Authenticate.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Environment+Authenticate.swift
+//  LoopKitUI
+//
+//  Created by Rick Pasetto on 8/3/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import LocalAuthentication
+import SwiftUI
+
+public typealias AuthenticationChallenge = (_ description: String, _ completion: @escaping (Result<Void, Error>) -> Void) -> Void
+public extension Result where Success == Void {
+    static var success: Result {
+        return Result.success(())
+    }
+}
+fileprivate struct UnknownError: Swift.Error { }
+
+private struct AuthenticationChallengeKey: EnvironmentKey {
+    static let defaultValue: AuthenticationChallenge = { authenticationChallengeDescription, completion in
+        let context = LAContext()
+        var error:NSError?
+        if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+            context.evaluatePolicy(.deviceOwnerAuthentication,
+                                   localizedReason: authenticationChallengeDescription,
+                                   reply: { (success, error) in
+                                    DispatchQueue.main.async {
+                                        assert(success || error != nil)
+                                        completion(success ? .success : .failure(error ?? UnknownError()))
+                                    }
+            })
+        } else {
+            assert(error != nil)
+            completion(.failure(error ?? UnknownError()))
```

Comment:
I'm not sure failure is the correct outcome if the policy can't be evaluated. What if the error is `biometryNotAvailable` or `biometryNotEnrolled`? If a user doesn't have auth set up, they should still be able to save their settings.

The precedent here (in bolusing) is to proceed as successful if the policy can't be evaluated.

Reply:
Good point.  I'll ask Tejasvi in Slack what the behavior should be.